Signal/src/view controllers/MessagesViewController.m

```diff
@@ -1988,21 +1989,21 @@ - (void)imagePickerController:(UIImagePickerController *)picker
              
              SignalAttachment *attachment = [SignalAttachment imageAttachmentWithData:imageData
                                                                               dataUTI:dataUTI];
-             if (!attachment ||
-                 [attachment hasError]) {
-                 DDLogWarn(@"%@ %s Invalid attachment: %@.",
-                           self.tag,
-                           __PRETTY_FUNCTION__,
-                           attachment ? [attachment errorMessage] : @"Missing data");
-                 failedToPickAttachment(nil);
-             } else {
-                 [self dismissViewControllerAnimated:YES
-                                          completion:^{
-                                              OWSAssert([NSThread isMainThread]);
-                                              
+             [self dismissViewControllerAnimated:YES
```

Comment:
One significant change - previously if users selected an invalid image in the image picker (camera roll), we would just ... do nothing. No error, not dismiss the image picker.  Now we dismiss the picker before we show the error alert.  This isn't a perfect experience, but a) it's better than doing nothing (which felt broken) and b) we'll soon support arbitrary attachments and significantly raise attachment sizes so this will happen very rarely.